Lib/filecmp.py

```diff
@@ -53,10 +53,10 @@ def cmp(f1, f2, shallow=True):
     s2 = _sig(os.stat(f2))
     if s1[0] != stat.S_IFREG or s2[0] != stat.S_IFREG:
         return False
-    if shallow and s1 == s2:
-        return True
-    if s1[1] != s2[1]:
+    if s1 != s2:
```

Comment:
It seems like if mtime is different but otherwise files are the same, current code will return True but this PR will change it to return False. That both breaks backwards compatibility and doesn't seem right to me.